governance/program/src/processor/process_set_realm_config.rs

```diff
@@ -0,0 +1,70 @@
+//! Program state processor
+
+use borsh::BorshSerialize;
+use solana_program::{
+    account_info::{next_account_info, AccountInfo},
+    entrypoint::ProgramResult,
+    pubkey::Pubkey,
+};
+
+use crate::{
+    error::GovernanceError,
+    state::realm::{assert_valid_realm_config_args, get_realm_data_for_authority, RealmConfigArgs},
+};
+
+/// Processes SetRealmConfig instruction
+pub fn process_set_realm_config(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    config_args: RealmConfigArgs,
+) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+
+    let realm_info = next_account_info(account_info_iter)?; // 0
+    let realm_authority_info = next_account_info(account_info_iter)?; // 1
+
+    let mut realm_data =
+        get_realm_data_for_authority(program_id, realm_info, realm_authority_info.key)?;
+
+    if !realm_authority_info.is_signer {
+        return Err(GovernanceError::RealmAuthorityMustSign.into());
+    }
+
+    assert_valid_realm_config_args(&config_args)?;
+
+    let realm_custodian = if config_args.use_custodian {
+        let realm_custodian_info = next_account_info(account_info_iter)?;
+        Some(*realm_custodian_info.key)
+    } else {
+        None
+    };
+
+    if config_args.use_council_mint {
+        let council_token_mint_info = next_account_info(account_info_iter)?;
+
+        // Council mint can only be at present set to none (removed) and changing it to other mint is not supported
+        // It might be implemented in future versions but it needs careful planning
+        // It can potentially open a can of warms like what happens with existing deposits or pending proposals
```

Comment:
Seems like a bit of a footgun -- if you have a council mint, and want to set the realm config, you have to keep sending in the council mint, and if you forget, it's unset forever.  Is that normal?

Reply:
Yeah, I agree it's a bit of footgun but I think there are some mitigation factors.
1)  My thinking was that you always set the full realm config and not it's individual parts to make it a single and simpler instruction. 
2) In future versions I would like to support changing the council (and not just removing), but just wanted to avoid the complexity for now since I don't think it's going to be a common case anyway. Also the assumption is that any changes to the realm would go through a proposal vote so the chances of removing the council by mistake are rather slim.